catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,355 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import static java.lang.String.format;
+import static org.codice.ddf.catalog.ui.query.suggestion.LatLon.from;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.codice.usng4j.BoundingBox;
+import org.codice.usng4j.CoordinateSystemTranslator;
+import org.codice.usng4j.UsngCoordinate;
+import org.codice.usng4j.impl.CoordinateSystemTranslatorImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A regex-based processor for identifying potential MGRS coordinate literals in a given input
+ * string despite deviations from the official spec. The general policy is to be as forgiving as
+ * possible without sacrificing determinism.
+ *
+ * <p>Due to the strict expectations of usng4j regarding input strings, some additional processing
+ * is done to "backfill" necessary zeros for occurrences of a {@link MgrsPartType#NUMERIC} string.
+ *
+ * @see #processNumericType(String) for more information on this "backfilling" behavior.
+ */
+@SuppressWarnings("squid:S1135" /* Ticket number provided on comment */)
+public class MgrsCoordinateProcessor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MgrsCoordinateProcessor.class);
+
+  private static final Pattern PATTERN_MGRS_ZONE = Pattern.compile("\\d\\d?[CDEFGHJKLMNPQRSTUVWX]");
```

Comment:
❓ Wouldn't it be more forgiving for the user to allow lowercase letters as well?